src/workbookManager.ts

```diff
@@ -24,18 +44,40 @@ export class WorkbookManager {
             templateFile === undefined
                 ? await JSZip.loadAsync(WorkbookTemplate.SIMPLE_QUERY_WORKBOOK_TEMPLATE, { base64: true })
                 : await JSZip.loadAsync(templateFile);
+        if (templateFile !== undefined && initialDataGrid !== undefined) {
```

Comment:
I think maybe you have a bug here? If initialData is defined but templateFile is undefined (It's a valid scenario IIUC) so in line 46 we will init templateFile and then we will get into this if?

maybe you need to put this if before the initialize of the template file?